engine/pkg/externalresource/broker/broker.go

```diff
@@ -170,8 +170,10 @@ func (b *DefaultBroker) OpenStorage(
 	var desc internal.ResourceDescriptor
 	if !exists {
 		desc, err = b.createResource(ctx, fm, projectInfo, workerID, resName)
+	} else if !options.cleanBeforeOpen {
+		desc, err = b.getPersistResource(ctx, fm, record, resName)
 	} else {
-		desc, err = b.getPersistResource(ctx, fm, record, resName, options)
+		desc, err = b.cleanPersistResource(ctx, fm, record, resName)
```

Comment:
Seems `cleanPersistResource` does the following work. Maybe we can rename this function and make it easier to understand.
- Clean existing files if they exist
- Re-create resource if possible.